autofs-5.0.6 - fix result null check in read_one_map()

From: Ian Kent <ikent@redhat.com>

Fix the check and reset to NULL of the LDAP library allocated result
within the loop to fetch paged query values.
---

 CHANGELOG             |    1 +
 modules/lookup_ldap.c |    7 ++++++-
 2 files changed, 7 insertions(+), 1 deletion(-)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -168,6 +168,7 @@
 - fix LDAP result leaks on error paths.
 - fix fix LDAP result leaks on error paths.
 - dont retry ldap connect if not required.
+- fix result null check in read_one_map().
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/modules/lookup_ldap.c
+++ autofs-5.0.5/modules/lookup_ldap.c
@@ -2431,8 +2431,10 @@ static int read_one_map(struct autofs_po
 
 		if (rv == LDAP_ADMINLIMIT_EXCEEDED ||
 		    rv == LDAP_SIZELIMIT_EXCEEDED) {
-			if (sp.result)
+			if (sp.result) {
 				ldap_msgfree(sp.result);
+				sp.result = NULL;
+			}
 			if (sp.cookie) {
 				ber_bvfree(sp.cookie);
 				sp.cookie = NULL;
@@ -2452,6 +2454,8 @@ static int read_one_map(struct autofs_po
 		if (rv != LDAP_SUCCESS || !sp.result) {
 			unbind_ldap_connection(ap->logopt, sp.ldap, ctxt);
 			*result_ldap = rv;
+			if (sp.result)
+				ldap_msgfree(sp.result);
 			if (sp.cookie)
 				ber_bvfree(sp.cookie);
 			free(sp.query);
@@ -2469,6 +2473,7 @@ static int read_one_map(struct autofs_po
 			return NSS_STATUS_NOTFOUND;
 		}
 		ldap_msgfree(sp.result);
+		sp.result = NULL;
 	} while (sp.morePages == TRUE);
 
 	debug(ap->logopt, MODPREFIX "done updating map");
